src/hooks/useBalance.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+import { Balance } from '@polkadot/types/interfaces';
+import { formatBalance } from '@polkadot/util';
+import BN from 'bn.js';
+import { useEffect, useState } from 'react';
+
+import { useSourceTarget } from '../contexts/SourceTargetContextProvider';
+
+type State = [string, Balance, boolean, string];
+
+const ZERO = new BN(0);
+
+const useBalance = (address: string, providedSi: boolean = false): State[] => {
```

Comment:
> In this case the logic to show the balances is the following:

I'm pretty sure this is incorrect. The entire codebase should not care what kind of account is selected. There is always only:
- `source` chain - the one we are going to send the transaction to
- `target` chain - the one that will have a message delivered and will perform dispatch.

When user selects an account in the dropdown, the correct `source/target` variables should be set accordingly, so when user selects Rialto:Alice, the `source` chain should be pointing to `Rialto`. When the user selects `Millau:Alice` the `source` chain should be pointing to `Millau`. Selecting accounts from different networks should cause the UI to switch `source/target` pair correctly.

Reply:
I think I've jumped to conclusion too early. So the code seems correct now and the logic described here seems correct too. I think the complication comes from the way we use hooks and how we treat accounts.

I wonder if we could always associate accounts with their chains. I.e. every time you receive `account` it's not only just account id, but it actually always have an `api` or at least `chainName` associated with it, so that if you use any util functions there is no confusion what api / chain to use.

Reply:
The logic i described is correct i think , because the purpose of this is to show as well the balances within the account select component. If you check the designs you will see that you have the accounts from both chains with their respective derived account balance. 
For the sake of consistency and how the code was built, when initially the select is open there is already a source and target defined.